recipes/default.rb

```diff
@@ -46,16 +46,26 @@
 jenkins_username = node['osrfbuild']['agent']['username']
 node_make_jobs = 3 # TODO: find a better way of handling make_jobs
 node_base_name = node['hostname'] == 'localhost' ? node['ipaddress'] : node['hostname']
-node_labels = [ node['osrfbuild']['agent']['labels'] ]
+node_labels = if node['osrfbuild']['agent']['labels']
+                node['osrfbuild']['agent']['labels'].dup
+              else
+                Array.new
+              end
```

Comment:
There's a way to do this in one line with Ruby's [safe navigation operator](https://ruby-doc.org/core-3.0.2/doc/syntax/calling_methods_rdoc.html#label-Safe+Navigation+Operator) but I hesitate to do so given that it's not a very common or at-a-glance readable expression. Open to other preferences though.

```suggestion
node_labels = node['osrfbuild']['agent']['labels']&.dup || Array.new
```

Reply:
new syntax is really great but I have no previous idea about the safe navigation operator. I'm inclined to leave as it is in the PR.